### PR TITLE
Issue#10 Initial suggested change for strongly typed relationships 

### PR DESCRIPTION
**Pull request by Budlee**
An suggested change to the relationships types to make them strongly typed.

### CODE REVIEW

**Comment by Budlee** [comment]
@rocketstack-matt and @jpgough-ms , this is just a sample to get thoughts on the changes of strongly typed relationships. If this is seen as a good idea I'll update the API Pattern example as well

**Review by jpgough-ms — CHANGES_REQUESTED** [review]
Overall I like this approach, but I think the `oneOf` will clarify what is strictly needed on a relationship

**Comment by aidanm3341** [comment]
I support this change! While building the visualiser, I had actually assumed that these were already strongly typed.

**Comment by aidanm3341** [comment]
I like baking these types into the schema - but if for any reason we didn't want to change the schema in this way, an alternative would be to create a spectral ruleset that checks for the existence of the required fields given some 'relationship-type'

File: calm/draft/2024-02/meta/core.json [hunk 1]
```diff
@@ -146,13 +135,70 @@
...
+        "destination": {
+          "type": "string"
+        }
+      }
+    },
+    "deployed-in-type": {
+      "type": "object",
+      "properties": {
+        "nodes": {
+          "type": "array",
```
[the remark this answers is not in the document]
Comment (Budlee) [reply calm/draft/2024-02/meta/core.json:175]: Like that a lot

File: calm/draft/2024-02/meta/core.json [hunk 2]
```diff
@@ -61,31 +61,21 @@
...
-              "type": "array",
-              "items": {
-                "type": "string"
-              }
+            "interacts": {
+              "$ref": "#/defs/interacts-type"
             },
-            "source": {
-              "type": "string"
+            "connects": {
```
[the remark this answers is not in the document]
Comment (Budlee) [reply calm/draft/2024-02/meta/core.json:69]: Yep, I'll apply that

File: calm/samples/traderx/traderx.json [hunk 3]
```diff
@@ -1,5 +1,5 @@
 {
-  "$schema": "https://raw.githubusercontent.com/finos-labs/architecture-as-code/main/calm/draft/2024-02/meta/calm.json",
+  "$schema": "../../../calm/draft/2024-02/meta/calm.json",
```
Comment (rocketstack-matt) [inline calm/samples/traderx/traderx.json:2]: Should remain full path

File: calm/samples/traderx/traderx.json [hunk 4]
```diff
@@ -118,153 +118,169 @@
         "nodes": [
           "web-client"
         ]
-      }
+      },
+      "relationship-t": ""
```
Comment (rocketstack-matt) [inline calm/samples/traderx/traderx.json:122]: Looks like a typo - should remove this line

File: calm/samples/traderx/traderx.json [hunk 5]
```diff
@@ -118,153 +118,169 @@
...
-        "nodes": [
-          "web-client",
-          "web-gui-process",
-          "position-service",
-          "traderx-db"
-        ],
-        "container": "internal-bank-network"
-      }
+      "relationship-type": {
+        "deployed-in": {
```
Comment (rocketstack-matt) [inline calm/samples/traderx/traderx.json:185]: Minor point but should we change the deployed-in to show the container first, does it read more naturally?

File: calm/samples/traderx/traderx.yaml [hunk 6]
```diff
@@ -1,4 +1,4 @@
-$schema: https://raw.githubusercontent.com/finos-labs/architecture-as-code/main/calm/draft/2024-02/meta/calm.json
+$schema:  ../../../calm/draft/2024-02/meta/calm.json
```
Comment (rocketstack-matt) [inline calm/samples/traderx/traderx.yaml:1]: Full path

**Review by rocketstack-matt — CHANGES_REQUESTED** [review]
I like the proposal, if you can get this tidied up and ready to push @Budlee I think we'll be happy to make this change.

**Comment by Budlee** [comment]
I’ll add these suggestions in. The bulk of it I completed yesterday and was just doing the tidy up. Send

File: calm/samples/traderx/traderx.json (hunk 3 above)
Comment (Budlee) [reply calm/samples/traderx/traderx.json:2]: reverted

**Comment by Budlee** [comment]
@rocketstack-matt , @aidanm3341  and @jpgough-ms I'd appreciate any additional feedback. All the suggestions have been mare

File: calm/draft/2024-02/meta/core.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (rocketstack-matt) [reply calm/draft/2024-02/meta/core.json:201]: I see we now have minItems, but don't we need it to be at least 1? In fact, in the case of nodes should it be at least 2 and in the case of relationships at least 1?

[the remark this answers is not in the document]
Comment (Budlee) [reply calm/draft/2024-02/meta/core.json:201]: Updated

**Review by rocketstack-matt — APPROVED** [review]

File: calm/draft/2024-02/meta/calm.json [hunk 7]
```diff
@@ -16,6 +16,6 @@
   "title": "Common Architecture Language Model (CALM) Schema",
   "allOf": [
     {"$ref": "https://json-schema.org/draft/2020-12/schema"},
-    {"$ref": "https://raw.githubusercontent.com/finos-labs/architecture-as-code/main/calm/draft/2024-02/meta/core.json"}
+    {"$ref": "core.json"}
```
[the remark this answers is not in the document]
Comment (Budlee) [reply calm/draft/2024-02/meta/calm.json]: This was reverted

**Comment by Budlee** [comment]
@jpgough-ms can you check the changes you requested, these should be done.

**Review by jpgough-ms — CHANGES_REQUESTED** [review]
@rocketstack-matt should this be in `draft/2024-03/meta` this is going to break patterns (as demonstrated by the need to change the samples).

**Comment by rocketstack-matt** [comment]
That's a very good point, I guess it should be, even by the guidelines we were discussing on the last call, but especially with it creating breaking change.

**Comment by rocketstack-matt** [comment]
@Budlee per @jpgough-ms's comment would you be able to raise this change as a new draft. That will also require that we version the samples so we have the old and new versions.

**Pull request closed without merging**